components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -614,3 +616,28 @@ sealed class ContainerAction : BrowserAction() {
      */
     data class RemoveContainerAction(val contextId: String) : ContainerAction()
 }
+
+/**
+ * [BrowserAction] implementations related to updating search engines in [SearchState].
+ */
+sealed class SearchAction : BrowserAction() {
+    /**
+     * Updates [BrowserState.search] to add/modify [SearchState.searchEngines].
+     */
+    data class UpdateSearchEngineListAction(val searchEngineList: List<SearchEngine>) : SearchAction()
+
+    /**
+     * Updates [BrowserState.search] to add/modify a custom [SearchEngine].
+     */
+    data class UpdateCustomSearchEngineAction(val searchEngine: SearchEngine) : SearchAction()
```

Comment:
Looking at this I wonder if we should rethink the `type` we introduced on `SearchEngine`. Maybe we should use actually different types (classes). 🤔  This will make it more obvious and safe: Here we say `UpdateCustomSearchEngine` but at the end of the day we accept all types of `SearchEngine` as parameter. And that can get confusing for the consuming app which may start to use the wrong actions? Not necessarily something for this PR, but maybe a follow-up?

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/7946 :).